File: base-v3/src/main/java/com/adyen/checkout/base/util/DateUtils.java
/*
 * Copyright (c) 2019 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by arman on 30/7/2019.
 */

package com.adyen.checkout.base.util;

import androidx.annotation.NonNull;

import com.adyen.checkout.core.exception.NoConstructorException;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.Locale;

@SuppressWarnings("PMD.UnsynchronizedStaticDateFormatter")
public final class DateUtils {

    private static final int FOUR_DIGIT = 4;
    private static final int START_OF_FOUR_DIGIT_INDEX = 2;
    private static final SimpleDateFormat SERVER_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd", Locale.US);

    private DateUtils() {
        throw new NoConstructorException();
    }

    /**
     * Convert 4 digit year to 2 digit.
     */
    @NonNull
    public static String removeFirstTwoDigitFromYear(@NonNull String fourDigitYear) {
        if (fourDigitYear.length() == FOUR_DIGIT) {
            return fourDigitYear.substring(START_OF_FOUR_DIGIT_INDEX, FOUR_DIGIT);
        }
        return fourDigitYear;
    }

    /**
     * Parse server date format.
     */
    @NonNull
    public static Calendar parseServerDateFormat(@NonNull String dateValue) {
        final Calendar calendar = Calendar.getInstance();
        try {
            final Date date = SERVER_DATE_FORMAT.parse(dateValue);
            calendar.setTime(date);
            return calendar;
        } catch (ParseException e) {
            return calendar;
        }
    }

    /**
     * Convert to server date format.
     */
    @NonNull
    public static String toServerDateFormat(@NonNull Calendar calendar) {
        return SERVER_DATE_FORMAT.format(calendar.getTime());
    }
}

File: base-v3/src/main/java/com/adyen/checkout/base/util/DateUtils.kt
/*
 * Copyright (c) 2020 Adyen N.V.
 *
 * This file is open source and available under the MIT license. See the LICENSE file for more info.
 *
 * Created by caiof on 27/11/2020.
 */

package com.adyen.checkout.base.util

import java.text.SimpleDateFormat
import java.util.Calendar
import java.util.Locale

class DateUtils private constructor() {
    companion object {
        @JvmStatic
        fun parseDateToView(month: String, year: String): String {
            // Refactor this to DateFormat if we need to localize.
            return if (year.length < 2) "$month/$year"
            else "$month/${year.substring(0..2)}"
        }

        /**
         * Convert to server date format.
         */
        @JvmStatic
        fun toServerDateFormat(calendar: Calendar): String {
            val serverDateFormat = SimpleDateFormat("yyyy-MM-dd", Locale.US)
            return serverDateFormat.format(calendar.time)
        }
    }
}
